src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -1100,21 +1108,73 @@ subroutine decomp_rates_mimics(bounds, num_soilc, filter_soilc, &
       mimics_cn_r = params_inst%mimics_cn_r
       mimics_cn_k = params_inst%mimics_cn_k
 
+      ! Local column-level ligninNratioAvg and annsum_npp_col because these are
+      ! not available when use_fates = .true. or when carbon_only = .true.
+      if (use_fates .or. cnallocate_carbon_only()) then
+         ! Initialize
+         do fc = 1,num_soilc
+            c = filter_soilc(fc)
+            ligninNratioAvg_local(c) = 0._r8
+            annsum_npp_col_local(c) = 0._r8
+            counter(c) = 0
+         end do
```

Comment:
The above loop is all I've really done for now:
I set local copies of `ligninNratioAvg` and `annsum_npp_col` to zero for FATES or carbon_only and these get used below.

If we decide to go with this solution for carbon_only and for FATES, then I will uncomment the next few lines, where the local copies of `ligninNratioAvg` and `annsum_npp_col` get values by pft.

Reply:
@ckoven thank you for this helpful pointer.

I've looked at this some more and found that I misspoke in one of my earlier posts:
Turns out that carbon_only has everything it needs and was failing for a different reason. So code mods in this PR are for FATES only. 

I plan to do the following (unless the group decides that we should go back to @ckoven 's original suggestion to make FATES and MIMICS mutually exclusive for now):
1) When running FATES, use annsum_npp from FATES in SoilBiogeochemDecompCascadeMIMICSMod.F90 as done in ch4Mod.
2) Follow @wwieder 's suggestion for ligninNratioAvg when running FATES.

Reply:
In today's commit/push I completed item (1).
Next I will address item (2).

Reply:
Latest commit addresses item (2).

Reply:
@rgknox and/or @glemieux one of you should validate that this is correct.